src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -60,12 +52,15 @@ internal class VisualStudioActiveDocumentTracker : ForegroundThreadAffinitizedOb
             : base(threadingContext, assertIsForeground: false)
         {
             _editorAdaptersFactoryService = editorAdaptersFactoryService;
-            _asyncTasks = new JoinableTaskCollection(threadingContext.JoinableTaskContext);
-            _asyncTasks.Add(ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+            ThreadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
             {
-                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
The default main thread priority for JTF.RunAsync calls in VS is background, which is pretty low. If you have any expectations about the rest of the code in this delegate happening before something else, you'll want to capture the `JoinableTask` returned from the `RunWithShutdownBlockAsync` call so you can await it later in front of those other operations that expect it to be done.

Reply:
This seems acceptable for our current purposes.